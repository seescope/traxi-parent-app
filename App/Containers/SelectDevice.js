import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { View, Image, TouchableOpacity } from 'react-native';
import { Actions } from 'react-native-router-flux';

import Background from '../Components/Background';
import HeaderText from '../Components/HeaderText';
import Spacing from '../Components/Spacing';
import BodyText from '../Components/BodyText';
import { TRAXI_BLUE } from '../Constants/Colours';
import { selectDevice } from '../Actions/Actions';

const style = {
  container: {
    backgroundColor: TRAXI_BLUE,
    flex: 1,
    padding: 8,
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  devicesContainer: {
    alignItems: 'flex-end',
    flexDirection: 'row',
  },
  deviceContainer: {
    marginHorizontal: 8,
  },
};

const SelectDevice = ({ parentName, onPress }) => (
  <Background style={style.container}>
    <HeaderText>What kind of device, {parentName}?</HeaderText>

    <Spacing />

    <BodyText>
      Tap one of the devices below to choose what kind of device your child is using.
    </BodyText>

    <Spacing height={36} />

    <View style={style.devicesContainer}>
      <TouchableOpacity onPress={() => onPress('phone')} style={style.deviceContainer}>
        <Image source={require('../Images/phone.png')} />
        <HeaderText>Phone</HeaderText>
      </TouchableOpacity>
      <TouchableOpacity onPress={() => onPress('iPad')} style={style.deviceContainer}>
        <Image source={require('../Images/tablet.png')} />
        <HeaderText>Tablet</HeaderText>
      </TouchableOpacity>
    </View>
  </Background>
);

SelectDevice.propTypes = {
  parentName: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  parentName: state.parentName,
});

const mapDispatchToProps = dispatch => ({
  onPress: deviceType => {
    if (deviceType === 'iPad') {
      dispatch(selectDevice('iPad'));
      Actions.createKid();
      return;
    }

    dispatch(selectDevice('unknown'));
    Actions.findKid();
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(SelectDevice);
