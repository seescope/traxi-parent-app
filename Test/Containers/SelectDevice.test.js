import { View } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { connect, Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import SelectDevice from '../../App/Containers/SelectDevice';

// Why? Why!?
global.Promise = require.requireActual('promise');

jest.mock('../../App/Actions/LoginWithMethod', () => {
  return jest.fn(() => {
    return (dispatch) => {
      return new Promise.resolve({ type: 'SOMETHIGN' });
    }
  });
});

jest.mock('react-native-router-flux', () => {
  mockFindKid = jest.fn();
  return {
    Actions: {
      findKid: mockFindKid
    }
  };
});

jest.mock('../../App/Components/Background');

const mockStore = configureStore([thunk]);
const testStore = mockStore({ parentName: 'Name'});

const SelectDeviceComponent = () => (
  <Provider store={testStore}>
    <SelectDevice />
  </Provider>
);

it('renders the <SelectDevice> component', () => {
  const tree = renderer.create(
    <SelectDeviceComponent />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
