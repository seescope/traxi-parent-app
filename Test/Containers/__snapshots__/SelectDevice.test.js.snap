exports[`test renders the <SelectDevice> component 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        animation="bounceInLeft"
        source={1}
        style={
          Object {
            "justifyContent": "flex-end",
            "marginHorizontal": 16,
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        animation="bounceInRight"
        source={1}
        style={
          Object {
            "justifyContent": "flex-end",
            "marginHorizontal": 16,
          }
        } />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 32,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-ExtraBold",
        "fontSize": 30,
        "marginBottom": -2,
        "textAlign": "center",
      }
    }>
    What kind of device?
  </Text>
  <View
    style={
      Object {
        "height": 8,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-Regular",
        "fontSize": 18,
        "fontWeight": "200",
        "textAlign": "center",
      }
    }>
    Tap one of the devices above to start monitoring.
  </Text>
</View>
`;
